requests/__init__.py

```diff
@@ -48,11 +48,20 @@
 __copyright__ = 'Copyright 2016 Kenneth Reitz'
 
 # Attempt to enable urllib3's SNI support, if possible
-try:
-    from .packages.urllib3.contrib import pyopenssl
-    pyopenssl.inject_into_urllib3()
-except ImportError:
-    pass
+# If the user has a recent version of OpenSSL, this
+# isn't needed as Python's core ssl module handles SNI, etc correctly.
+# If the requirements of urllib3.contrib.pyopenssl change,
+# this version should be modified appropraitely.
+#
+# See https://github.com/kennethreitz/requests/issues/3213 as well.
+import ssl
+
+if hasattr(ssl, 'OPENSSL_VERSION_INFO') and getattr(ssl, 'OPENSSL_VERSION_INFO') < (1, 0, 1):
```

Comment:
Why not more simply:

```py
_OPENSSL_VERSION = getattr(ssl, 'OPENSSL_VERSION_INFO', None)
if _OPENSSL_VERSION is None or _OPENSSL_VERSION < (1, 0, 1):
    try:
        # ...
    except ImportError:
        pass
```

This then will import pyOpenSSL on 2.6 *or* in situations where the built-in OpenSSL isn't quite so great.